internal/model/labels.go

```diff
@@ -59,6 +61,53 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	for key, localV := range l {
```

Comment:
We should also check `len(l) == len(labels)`, otherwise one may be a superset of the other.